crates/search/src/buffer_search.rs

```diff
@@ -2800,22 +2796,22 @@ mod tests {
             );
 
             search_bar.deploy(&deploy, window, cx);
-            assert_eq!(
-                search_bar.configured_options,
-                SearchOptions::CASE_SENSITIVE,
-                "Should have configured search options matching the settings"
-            );
             assert_eq!(
                 search_bar.search_options,
                 SearchOptions::REGEX | SearchOptions::WHOLE_WORD,
                 "Toggling a non-dismissed search bar with custom options should not change the default options"
             );
             search_bar.dismiss(&Dismiss, window, cx);
             search_bar.deploy(&deploy, window, cx);
+            assert_eq!(
+                search_bar.configured_options,
+                SearchOptions::CASE_SENSITIVE,
+                "After a settings update, dismissing the bar should update the configured options"
+            );
             assert_eq!(
                 search_bar.search_options,
                 SearchOptions::CASE_SENSITIVE,
-                "After hiding and showing the search bar, default options should be used"
+                "After a settings update, dismissing the bar should update the active search options"
```

Comment:
The wording here is odd: dismissing per se does not update anything, does it?
The toggling / "hiding and showing" does, right?

Reply:
Dismissing the search bar is the action that actually updates the search options, not necessarily showing it again. But that's probably an implementation detail that doesn't need to be reflected in the wording. I'll take another shot at it now